src/tensors/cpu/device.cpp

```diff
@@ -17,6 +17,15 @@ namespace cpu {
 // functions which tests for alignment. 
 #ifdef _WIN32
 #define MALLOC(size) _aligned_malloc(size, alignment_)
+#elif __APPLE__
+//On macos, aligned_alloc is available only on c++17
+//Furthermore, it requires that the memory requested is an exact multiple of the alignment, otherwise it fails
+void * apple_aligned_alloc(size_t alignment_, size_t size) {
```

Comment:
Is posix_memalign the same as aligned_alloc on Linux? Then we should use it in both cases, and condition on the correct #define (something POSIX I presume)

Reply:
`aligned_alloc` is the same as `posix_memalign` if you are using the standard C++ libraries. I believe, if you use `libc++` it might be different. At any rate, `aligned_alloc` is stricter than `posix_memalign` and you can definitely use it on linux. If you prefer, I can merge those two codepaths. I do find `aligned_alloc` more C++ style.

Reply:
@frankseide Take a look at what I just pushed.

Reply:
We also need to update: https://github.com/marian-nmt/FBGEMM/blob/84e66a976046180187724aff60a236c5378fde7c/include/fbgemm/Utils.h#L14 
as this won't work on clang with libc++. @ykim362 , why was the allocator changed for fbgemm? Is it just for Windows support?

Reply:
I prefer the unified code path over the C++ style. Please add a comment aht posix_memalign is used instead of aligned_alloc for clang compatibility